api/app/GraphQL/Mutations/CreateUserSkill.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace App\GraphQL\Mutations;
+
+use App\Models\ExperienceSkill;
+use App\Models\UserSkill;
+use Nuwave\Lighthouse\Exceptions\ValidationException;
+
+final class CreateUserSkill
+{
+    /**
+     * Create a UserSkill model or restore it
+     * @param  null  $_
+     * @param  array{}  $args
+     * @return UserSkill
+     */
+    public function __invoke($_, array $args)
+    {
+        // check if the model exists accounting for trashed and then restore or error out, if not found proceed to create
+        $existingModel = UserSkill::where('user_id', $args['user_id'])
+            ->where('skill_id', $args['skill_id'])
+            ->withTrashed()
+            ->first();
+        if ($existingModel !== null) {
+            if (($existingModel->deleted_at !== null)) {
+                $existingModel->restore();
+                ExperienceSkill::where('user_skill_id', $existingModel->id)->restore(); // restore records on experience_skill
+                return $existingModel;
+            }
+            throw ValidationException::withMessages(["DuplicateUserSkill"]);
+        }
+
+        $createdModel = UserSkill::create([
+            'user_id' => $args['user_id'],
+            'skill_id' => $args['skill_id'],
+            'skill_level' => isset($args['skill_level']) ? $args['skill_level'] : null,
```

Comment:
Is this necessary? Won't it default to null if it's not set?

Reply:
I had errors popping up while writing testing that this change in https://github.com/GCTC-NTGC/gc-digital-talent/pull/7579/commits/1a055ddf7f5c6a7fdef767a3b06622695e29e5b8 seemed to resolve
So, maybe, maybe not? 🤷